src/feature/client/entrynodes.c

```diff
@@ -3184,13 +3237,22 @@ entry_guards_load_guards_from_state(or_state_t *state, int set)
       tor_assert(gs);
       smartlist_add(gs->sampled_entry_guards, guard);
       guard->in_selection = gs;
+      /* Recompute the next_sampled_id from the state  */
+      if (gs->next_sampled_idx <= guard->sampled_idx) {
+        gs->next_sampled_idx = ++guard->sampled_idx;
```

Comment:
Wait -- why are you incrementing guard->sampled_idx here?  Don't you just want to say "guard->sampled_idx + 1"?

Reply:
Wait -- yes. Now I wonder why a test did not catch it. Looking into it.